test/schemas/compilation-result.coffee

```diff
@@ -29,18 +29,20 @@ module.exports = (options = {}) ->
   # will default to true (= structure must contain transaction names and paths)
   paths = if options.paths is false then false else true
 
+  headersSchema =
+    type: 'array'
+    items:
+      type: 'object'
+      properties:
+        name: {type: 'string'}
+        value: {type: 'string'}
+
   requestSchema =
     type: 'object'
     properties:
       uri: {type: 'string', pattern: '^/'}
       method: {type: 'string'}
-      headers:
-        type: 'object'
-        patternProperties:
-          '': # property of any name
-            type: 'object'
-            properties:
-              value: {type: 'string'}
+      headers: headersSchema
```

Comment:
Perhaps it would make sense to use JSON Schema referencing since this is JSON schema. But it's down to you, I don't think it matters either way.

```yaml
headers:
  '$ref': '#!/definitions/headers'
definitions:
  headers:
    type: array
    ....
```

Outside of the scope of this PR, but it might be nice to pull this out into a separate file. (I made a tool for testing schemas in JSON once that could be a neat idea here: https://github.com/kylef/jsonschema-test)

Reply:
The main/only reason this is not in `.json` or `.schema` files (not sure what would be the appropriate extension) is that they're parametrized. So I generate them accroding to needs of a particular test.